src/Symfony/Component/HttpKernel/CHANGELOG.md

```diff
@@ -7,6 +7,7 @@ CHANGELOG
  * Add method `isKernelTerminating()` to `ExceptionEvent` that allows to check if an exception was thrown while the kernel is being terminated
  * Add `HttpException::fromStatusCode()`
  * Add `$validationFailedStatusCode` argument to `#[MapQueryParameter]` that allows setting a custom HTTP status code when validation fails
+ * Improve error reporting when targeting the wrong Request class in a controller action
```

Comment:
I'm not sure if this entry is needed, let me know if not

Reply:
the real new feature is the new exception so we could focus on that one in this message